ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -175,12 +298,41 @@ object EndpointsCompanion {
 
   private[http] def format(a: JsValue): ByteString = ByteString(a.compactPrint)
 
-  private[http] def decodeAndParsePayload[A](jwt: Jwt, decodeJwt: ValidateJwt)(implicit
+  private[http] def legacyDecodeAndParsePayload[A](jwt: Jwt, decodeJwt: ValidateJwt)(implicit
       parse: ParsePayload[A]
-  ): Unauthorized \/ (Jwt, A) = {
+  ) =
     for {
       a <- decodeJwt(jwt): Unauthorized \/ DecodedJwt[String]
       p <- parse.parsePayload(a)
     } yield (jwt, p)
+
+  private[http] def decodeAndParsePayload[A](
+      jwt: Jwt,
+      decodeJwt: ValidateJwt,
+      userManagementClient: UserManagementClient,
+      ledgerIdentityClient: LedgerIdentityClient,
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      legacyParse: ParsePayload[A],
+      createFromUserToken: CreateFromUserToken[A],
+      fm: Monad[Future],
+      ec: ExecutionContext,
+  ): ET[(Jwt, A)] = {
+    for {
+      a <- EitherT.either(decodeJwt(jwt): Unauthorized \/ DecodedJwt[String])
+      p <- legacyParse
```

Comment:
have you considered following the approach in the ledger API of making the decision based on whether `sub` is present.